R/derive_vars_dt.R

```diff
@@ -165,27 +165,70 @@
 #'   dtc = MHSTDTC,
 #'   highest_imputation = "M",
 #'   date_imputation = "04-06"
-#' )
-#' @caption Avoid imputation before a user-defined date (`min_dates`)
+#'   )
+#'
+#' @caption Applying a lower boundary to date imputation with (`min_dates`)
 #' @info In this example, we derive `ASTDT` where `AESTDTC` is all partial dates in
 #' need of imputation. Using `min_dates = exprs(TRTSDTM)`, we are telling the function
-#' to not allow imputation dates to be before the treatment start date
-#' via `min_dates` argument. Note that the second record does not get imputed
-#' as it is before `TRTSDTM`.
+#' to apply the treatment start date `(TRTSDTM)` as a lower boundary for imputation
+#' via the `min_dates` argument.
+#' This means:
+#' - For partial dates that could potentially include `TRTSDTM` (case 1 & 2),
+#' the imputed date is adjusted to `TRTSDTM`
+#' - For partial dates that are entirely before `TRTSDTM` (case 3 & 4),
+#' standard imputation rules apply without adjustment
+#' - For partial dates that are entirely after `TRTSDTM` (case 5),
+#' standard imputation rules apply
```

Comment:
maybe a little crazy but could the Case 1-5 be added as variable for the examples